add mesh expansion to the deployment models document (#6209)
istio/istio.io: content/en/docs/ops/deployment/deployment-models/index.md

  - trust
  - single-mesh
  - multiple-meshes

>   - mesh-expansion

Reviewer: ```suggestion
  - virtual-machines
```

Reviewer 2: I don't think "mesh expansion" is the right phrase ... nobody knows what that means initially :). We should say VMs, or something like that.

If there is more than one cluster being deployed, and more specifically in isolated networks,
are they going to be connected into a single {{< gloss >}}multicluster{{< /gloss >}}
service mesh or will they be federated into a {{< gloss >}}multi-mesh{{< /gloss >}} deployment?

> Will services on virtual machines or bare metal computers be connected to the mesh?

Reviewer: ```suggestion
Will services running on virtual machines or bare metal computers be part of the mesh?
```

> Will services running on virtual machines or bare metal computers be part of the mesh?

Author: @rcaballeromx the questions in the list above are in the future tense, the reader must decide how to create a mesh, for example: _...are they going to be connected into a single multicluster
service mesh or will they be federated into a multi-mesh..._.

So the question is: "do you want your mesh to include workloads running on VMs?" It is up to you, (the reader), you must decide.

> Will services running on virtual machines or bare metal computers be part of the mesh?

Reviewer 3: They ask themselves the questions and decide in their present @vadimeisenbergibm 
This is a question of style and I don't feel strongly either way. You can leave it in future tense but please avoid using the term service:
```suggestion
Will you add workloads running on virtual machines or bare metal systems to the mesh?
```

> Will services running on virtual machines or bare metal computers be part of the mesh?

Reviewer: Re: services vs. workloads. This isn't just a case of some workloads implementing a service running on a VM. We are talking about integrating into the mesh a "service" that is fully deployed on a VM, so I think service is the right term to use here.

That said, as I said in my other comment, this whole thing should move to a separate doc.

All of these questions, among others, represent independent dimensions of configuration for an Istio deployment.

1. single or multiple cluster
1. single or multiple network
1. single or multiple control plane
1. single or multiple mesh

> 1. mesh expansion (integration with VMs/bare metal)

Reviewer: ```suggestion
1. Integration with VMs/bare metal
```

When each team has their own mesh, cross-mesh communication follows the
concepts described in the [multiple meshes](#multiple-meshes) model.

## Virtual machines and bare metal

> In same cases you might have to deploy some of the services on virtual machines or bare metal computers. The

Author: This sentence says that sometimes you must deploy some workloads on VMs. This statement is not related to the Istio support of VMs.

> In same cases you might have to deploy some of the services on virtual machines or bare metal computers. The

Reviewer 3: @vadimeisenbergibm That's then a redundant statement. What's missing here is what impact does adding VMs and bare metal systems have on how readers are deploying the mesh. Readers of this doc know whether or not they need to add VMs or bare metal systems to the mesh. What is your goal in stating vaguely that "sometimes" this is needed? What are those times? Under what circumstances? With what goals? The answers to those questions vary from reader to reader.  On this doc we are trying to focus on helping folks determine what deployment suits their needs, not on encouraging them to use any specific deployment model.

> In same cases you might have to deploy some of the services on virtual machines or bare metal computers. The

Author: I specify the reasons for running workloads on VMs below:
```
- Special hardware requirements
- Tighter isolation
- Tighter integration between the workload and the OS
- Legacy systems, workloads, or deployments
```

> reasons for deployments on VMs/bare metal computers could be special hardware requirements, tighter isolation, tighter integration with the host's OS, legacy deployments.

Author: See https://github.com/istio/istio.io/pull/6209#discussion_r364678463. I am OK with making the sentence as list.

> reasons for deployments on VMs/bare metal computers could be special hardware requirements, tighter isolation, tighter integration with the host's OS, legacy deployments.

Reviewer 3: My edits go towards providing readers with the reasons they should consider a deployment model that includes VMs or bare metal systems. I would be ok if you want to change the introductory sentence of the list to be more explicit:
`Chose a deployment model that adds VMs or bare metal systems to the mesh if any of the following apply:`

Having some of the services on VMs/bare metal, you may want to connect the services into a mesh and benefit from all the

> service mesh features: traffic management, security, observability, and others. You can deploy the Istio control plane(s)

Author: @rcaballeromx please note that the goal is to present Istio features in general, since this is a high-level statement. The features of Istio on general: traffic management, security, observability. The list of the Istio features is too detailed to present the reader in a high level description.

> service mesh features: traffic management, security, observability, and others. You can deploy the Istio control plane(s)

Reviewer 3: The problem is that your list is not complete. This is also not the place to present Istio features in general. Sometimes, less is more. As @frankbu and @smawson stated, this doc is not about showcasing why folks should add VMs to their deployments.

> service mesh features: traffic management, security, observability, and others. You can deploy the Istio control plane(s)

Author: Note that we do not know if all the features of Istio work for VMs. So we cannot state that the users will benefit from all the Istio features when they integrate VMs with the mesh. However, we still want to provide users some motivation to integrate VMs with the mesh. So what should be stated here: "you will benefit from some of the features of Istio"? Or "you will benefit from most of the features"?

>  this doc is not about showcasing why folks should add VMs to their deployments

The reader could ask a reasonable question: what do I gain by integrating VMs with the mesh?

on some of your clusters and integrate your VMs/bare metal computers with the mesh.

Adding a VM/bare metal computer to a mesh means installing an Istio sidecar and a Node agent on the machine and

> configuring them to communicate with the control plane(s) on some of the clusters. Once you add a VMs/bare metal

Reviewer 3: ```suggestion
1. Configure the sidecar and the agent to communicate with the control plane or planes in your deployment.
```

> computers to a mesh, you can configure the control plane(s) of the mesh to control communication between services on the

Reviewer 3: ```suggestion
1. Configure the control plane or planes of your mesh to control communication between the workloads running on 
    the clusters, running on the VMs, and running on the bare metal systems, for example:
```

> clusters and on the machines (from a cluster to a VM, from a VM to a cluster, from a VM to another VM in the mesh).

Reviewer 3: ```suggestion
    - From a cluster to a VM
    - From a VM to a cluster
    - From a VM to another VM in the mesh
```

When each team has their own mesh, cross-mesh communication follows the
concepts described in the [multiple meshes](#multiple-meshes) model.

> ## Mesh expansion

Reviewer: I think we decided not to use the term "Mesh Expansion" because it's not a known industry term and any configuration that adds services to the mesh would be considered "expanding the mesh".
```suggestion
## Virtual machines and bare metal
```

In same cases you might have to deploy some of the services on virtual machines or bare metal computers. The
reasons for deployments on VMs/bare metal computers could be special hardware requirements, tighter isolation, tighter integration with the host's OS, legacy deployments.

Having some of the services on VMs/bare metal, you may want to connect the services into a mesh and benefit from all the
service mesh features: traffic management, security, observability, and others. You can deploy the Istio control plane(s)

> on some of your clusters and expand the mesh of the cluster(s) by adding VMs/bare metal computers into the mesh.

Reviewer 2: Again, I'd lose the phrase "expand the mesh". Let's go with Frank's phrasing of "integrate with...".

Author: Addressed in https://github.com/istio/istio.io/pull/6209/commits/26fc6d5dcbe19d51ecd80c46190beedf0aec6456.

> The VMs/bare metal computers can share the same network with the clusters of the mesh, or they can be on

Reviewer 2: This paragraph seems redundant with other concepts. I think we can lose it.

Author: Done in https://github.com/istio/istio.io/pull/6209/commits/3f4ce88594fb82b137a62288eea4f7a30f72a0de.
